### PR TITLE
Issue #3402: Standardise the display of labels and machine names.

### LINKED ISSUE #3402: [UX] Standardize Label and `machine name` usage in admin listings
## Describe your issue or idea

In https://github.com/backdrop/backdrop-issues/issues/3395 the `machine name` value was moved into the `View name` column:

screen shot 2018-11-30 at 2 57 31 pm

I've followed that pattern in [recent changes in the Paragraphs module in contrib](https://github.com/backdrop-contrib/paragraphs/issues/23#issuecomment-443291869), and I propose it would be good to use it more broadly in core and make things consistent. We'd first need to get a list of all the pages that list `machine name` as a separate column, but for starters here are some suggestions/questions:

 - all the `Manage fields` pages have separate columns (on content types, user account fields, taxonomy vocabulary fields)
screen shot 2018-11-29 at 5 30 11 pm

 - Content types list machine name after the label but include the words "Machine name: " which could maybe be removed:
screen shot 2018-11-30 at 3 05 38 pm

 - On the other hand, Taxonomy vocabularies don't list the machine name at all (although it can be useful to have):
screen shot 2018-11-30 at 3 11 27 pm

---

**Final interfaces after the PR:**

Content types list:
Screen Shot 2021-08-29 at 3 48 28 PM

Taxonomy vocabularies:
Screen Shot 2021-08-29 at 3 48 49 PM

Custom blocks:
Screen Shot 2021-08-29 at 3 50 15 PM

Manage displays:
Screen Shot 2021-08-29 at 3 51 08 PM

File types:
Screen Shot 2021-08-29 at 3 51 08 PM

### PR DESCRIPTION
**Pull request by klonos**
Fixes https://github.com/backdrop/backdrop-issues/issues/3402

Replaces https://github.com/backdrop/backdrop/pull/3473

### CODE REVIEW

File: core/themes/seven/css/style.css [hunk 1]
```diff
@@ -1900,3 +1900,21 @@ div.add-or-remove-shortcuts {
 @keyframes animate-stripes {
   0% { background-position: 0 0; } 100% { background-position: 50px 50px; }
 }
+
+/**
+ * Labels & machine names.
+ */
+.label {
```
Comment (jenlampton) [inline core/themes/seven/css/style.css]: This selector is too general, it also catches all instances of `label` that are not in the name column of an admin table. See for example: https://pr3710-9c7m7y6ecoadgavjmgxi3wih3qnrtmfq.tugboat.qa/node/add 
Screen Shot 2021-08-27 at 3 37 41 PM

File: core/themes/seven/css/style.css [hunk 2]
```diff
@@ -1900,3 +1900,21 @@ div.add-or-remove-shortcuts {
...
+  font-weight: bold;
+  padding-left: 28px; /* LTR */
```
Comment (jenlampton) [inline core/themes/seven/css/style.css]: This padding is showing up everywhere, even for labels that are not in draggable rows. See https://pr3710-9c7m7y6ecoadgavjmgxi3wih3qnrtmfq.tugboat.qa/admin/structure/types 
Screen Shot 2021-08-27 at 3 40 47 PM

and 
https://pr3710-9c7m7y6ecoadgavjmgxi3wih3qnrtmfq.tugboat.qa/admin/structure/block
Screen Shot 2021-08-27 at 3 42 37 PM

Comment (klonos) [reply core/themes/seven/css/style.css]: Thanks @jenlampton ...I've fixed this by adding a `tabledrag-handles-shown` class to the table (and hiding that when the "show weights" link is clicked).

File: core/themes/seven/css/style.css (hunk 1 above)
Comment (jenlampton) [reply core/themes/seven/css/style.css]: looks better!

**Comment by klonos** [comment]
The links to the details of the Zen CI tests 404 😞 

Closing and re-opening, to trigger the tests again...

**Comment by klonos** [comment]
Zen doesn't like me 😓

File: core/modules/field_ui/field_ui.theme.inc [hunk 3]
```diff
@@ -115,8 +117,11 @@ function theme_field_ui_view_modes($variables) {
 
   // Add the row for the default display mode.
   $row = array();
-  $row['label'] = $label . '<div class="description">' . $label_desc . '</div>';
-  $row['machine'] = t('default');
+  $row['label'] = theme('label_machine_name__node_display__' . $bundle_machine_name, array(
+    'label' => $label,
+    'machine_name' => t('default'),
```
Comment (klonos) [inline core/modules/field_ui/field_ui.theme.inc]: This was here before, but if it is a machine name, then should we be translating it?

File: core/misc/tabledrag.js [hunk 4]
```diff
@@ -216,8 +216,9 @@ Backdrop.tableDrag.prototype.addColspanClass = function(columnIndex) {
 Backdrop.tableDrag.prototype.hideColumns = function () {
   // Hide weight/parent cells and headers.
   $('.tabledrag-hide', 'table.tabledrag-processed').css('display', 'none');
+  $('table.tabledrag-processed').addClass('tabledrag-handles-shown');
   // Show TableDrag handles.
-  $('.tabledrag-handle', 'table.tabledrag-processed').css('display', '');
```
Comment (jenlampton) [inline core/misc/tabledrag.js:220]: is this change related to the label/machine update? I think it might have slipped in from our other tabledrag issue today.

File: core/misc/tabledrag.js [hunk 5]
```diff
@@ -240,7 +241,8 @@ Backdrop.tableDrag.prototype.hideColumns = function () {
  */
 Backdrop.tableDrag.prototype.showColumns = function () {
   // Show weight/parent cells and headers.
-  $('.tabledrag-hide', 'table.tabledrag-processed').css('display', '');
```
Comment (jenlampton) [inline core/misc/tabledrag.js:243]: is this change related to the label/machine update? I think it might have slipped in from our other tabledrag issue today.

File: core/modules/system/css/system.theme.css [hunk 6]
```diff
@@ -544,3 +544,10 @@ img.align-right {
   display: table;
   clear: both;
 }
+
+/**
+ * Labels & machine names.
+ */
+.machine-name {
+  font-size: smaller;
```
Comment (jenlampton) [inline core/modules/system/css/system.theme.css:559]: Do we use `smaller` elswehere in `system.theme.css`?

File: core/modules/user/user.theme.inc [hunk 7]
```diff
@@ -172,15 +172,18 @@ function theme_user_permission_description($variables) {
...
   foreach (element_children($form['roles']) as $role_name) {
-    $label = $form['roles'][$role_name]['#role']->label;
     $row = array();
+    $row[] = theme('label_machine_name__role', array(
+      'label' => $form['roles'][$role_name]['#role']->label,
+      'machine_name' => $form['roles'][$role_name]['#role']->name,
+    ));
     if (in_array($role_name, array(BACKDROP_ANONYMOUS_ROLE, BACKDROP_AUTHENTICATED_ROLE))) {
-      $row[] = t('@name <em>(required)</em>', array('@name' => $label));
+      $row[] = t('yes');
```
Comment (jenlampton) [inline core/modules/user/user.theme.inc]: Should this be captialized?

File: core/themes/seven/css/style.css [hunk 8]
```diff
@@ -1900,3 +1900,21 @@ div.add-or-remove-shortcuts {
...
+/**
+ * Labels & machine names in admin listing tables.
+ */
+table .label {
+  font-weight: bold;
+}
+table .machine-name {
+  color: #666;
+}
+.tabledrag-handles-shown .label,
```
Comment (jenlampton) [inline core/themes/seven/css/style.css:1934]: looks great!

Comment (klonos) [reply core/themes/seven/css/style.css:1934]: Thanks 🙂

File: core/misc/tabledrag.js (hunk 4 above)
Comment (klonos) [reply core/misc/tabledrag.js:220]: This code attempts to remove the `display: none`, but results in an "empty" `style` attribute in the element (similar to what we resolved in https://github.com/backdrop/backdrop-issues/issues/1849 with empty classes left behind). I can remove it, but does it worth its own issue?

File: core/misc/tabledrag.js (hunk 5 above)
Comment (klonos) [reply core/misc/tabledrag.js:243]: See my reply to the previous comment. This is to avoid "empty" `style` attributes left behind, but I've removed it.

File: core/modules/user/user.theme.inc (hunk 7 above)
Comment (klonos) [reply core/modules/user/user.theme.inc]: Not too fussed about it TBH. I would really like us to change this to a useful description instead (https://github.com/backdrop/backdrop-issues/issues/5179), which would replace this anyway.

File: core/modules/system/css/system.theme.css (hunk 6 above)
Comment (klonos) [reply core/modules/system/css/system.theme.css:559]: We use `font-size: smaller;` in 3 files in core:
- views_ui module: https://github.com/backdrop/backdrop/blob/1.x/core/modules/views_ui/css/views_ui.admin.theme.css#L307
- system module: https://github.com/backdrop/backdrop/blob/1.x/core/modules/system/css/system.maintenance.css#L9
- seven theme:
  - https://github.com/backdrop/backdrop/blob/1.x/core/themes/seven/css/seven.base.css#L93
  - https://github.com/backdrop/backdrop/blob/1.x/core/themes/seven/css/seven.base.css#L97
  - https://github.com/backdrop/backdrop/blob/1.x/core/themes/seven/css/seven.base.css#L102

File: core/misc/tabledrag.js (hunk 4 above)
Comment (klonos) [reply core/misc/tabledrag.js:220]: ...never mind. I've removed this, and filed a PR with a slightly improved fix in https://github.com/backdrop/backdrop-issues/issues/5188

File: core/modules/user/user.theme.inc (hunk 7 above)
Comment (quicksketch) [reply core/modules/user/user.theme.inc]: I don't think this is a pattern we've used elsewhere of saying "yes" in a column for requiredness:

image

Just using "yes" seems vague to me what it means. I think putting "Required" in each column may be more clear.

This column possibly should also be given `priority-low` so it hides on mobile.

Comment (klonos) [reply core/modules/user/user.theme.inc]: As discussed with @quicksketch in the sprint zoom room, the preference here is to revert to adding the `(required)` bit as a suffix to the role name, and remove the "Required" column altogether. In a follow up like https://github.com/backdrop/backdrop-issues/issues/5179 we can add the ability to provide descriptions for roles, and have the descriptions for Anonymous/Authenticated be "locked" or hardcoded to the description text we provide.

Doing that next...

Comment (klonos) [reply core/modules/user/user.theme.inc]: Done (see https://github.com/backdrop/backdrop/pull/3710/commits/0fa3f3606f8a9f6def49ffacbe2923b4d58ad1dd). Side-by-side before/after:

image

File: core/modules/system/css/system.theme.css (hunk 6 above)
Comment (klonos) [reply core/modules/system/css/system.theme.css:559]: @jenlampton I believe that this is the last remaining concern for this PR now. Please let me know if you would like that changed to a specific px/em value instead.

Comment (jenlampton) [reply core/modules/system/css/system.theme.css:559]: No, since it's in `/core/modules/system/css/system.maintenance.css` It's a pattern we use in core (and not only in Seven) --- I just hadn't seen it before!

(Views could have been a mistake, just inherited from contrib)

**Comment by klonos** [comment]
image
🙄

**Comment by jenlampton** [comment]
LGTM!

**Comment by klonos** [comment]
Our old friend `Start and end date on same day with timezone` strikes again 😞

**Comment by klonos** [comment]
...I honestly don't know why the deploy scripts fail 🤷🏼 ...tests seem to be passing, so I think I'll give up. If someone else wants to keep closing/reopening this till Zen CI behaves, then please be my guest 😓

**Comment by laryn** [comment]
Thanks for the work that went into this PR @klonos, @jenlampton, @BWPanda and @quicksketch! I think the result really helps the consistency and professionalism of the admin interface.

Merging into `1.20`.
